task_3/module/sg/variables.tf

```diff
@@ -0,0 +1,75 @@
+variable "create_sg" {
+  description = "value to detect if sg should be created"
+  type        = bool
+  default     = false
+}
+
+variable "security_group_name" {
+  description = "security group name"
+  type        = string
+  default     = "my_sg"
+}
+
+variable "vpc_id" {
+  description = "vpc to place an EC2 instance"
+  type        = string
+  default     = "vpc-00dc32518740e437f"
+}
+
+variable "tags" {
+  description = "tags"
+  type        = map
+  default = {
+    First_name = "firstName"
+    Last_name  = "secondName"
+  }
+}
+
+variable "sgr_ingress" {
+  description = "security group rules - ingress"
+  type        = list(map(string))
+  default = [
+    {
+      from_port   = "22"
+      to_port     = "22"
+      protocol    = "tcp"
+      cidr_blocks = "0.0.0.0/0"
```

Comment:
Please remove 0.0.0.0/0 from default values. This can cause a security issue.

Reply:
host provider is used to detect currect terraform host public ip:
```terraform
data "http" "myip" {
  url = "http://ipv4.icanhazip.com"
}
```
CIDR definition is changed accordingly
```terraform
resource "aws_security_group_rule" "my_sg_ingress" {
  type = "ingress"
  ...
  cidr_blocks       = ["${chomp(data.http.myip.body)}/32"]
  ...
}
```